meshers/blocky/voxel.cpp

```diff
@@ -224,12 +230,52 @@ static void bake_cube_geometry(Voxel &config, Voxel::BakedData &baked_data, int
 	CRASH_COND(atlas_size <= 0);
 	const float s = 1.0 / atlas_size;
 
+	std::vector<Vector2> deltaUV;
+
 	for (unsigned int side = 0; side < Cube::SIDE_COUNT; ++side) {
 		baked_data.model.side_uvs[side].resize(4);
 		std::vector<Vector2> &uvs = baked_data.model.side_uvs[side];
 		for (unsigned int i = 0; i < 4; ++i) {
 			uvs[i] = (config.get_cube_tile(side) + uv[i]) * s;
 		}
+
+		//Tangent calculation modeled after http://www.opengl-tutorial.org/intermediate-tutorials/tutorial-13-normal-mapping/#computing-the-tangents-and-bitangents
```

Comment:
I'm not very familiar with this, but it looks very complicated for a simple cube. The code here uses lookup tables with precalculated values, and UVs are deterministic as well, so why not just do that?

Reply:
I've changed it to that. I dumped the normalized tangents and stored one for each face in the cube_tables.

Reply:
I think you're right. Tangents typically follow the direction of the texture to avoid ugly edges and I was worried because the UVs are dependent on the atlas size we'd run into issues. But s will always be positive, so the UVs won't ever flip directions. Which means our normalized tangents should never change? 

I switched to a lookup table in the cube tables file.